### PR TITLE
Separate encrypted repository plugin integ tests

### PR DESCRIPTION
**Pull request by rjernst**
The encrypted repository module supports encryption with existing
repository plugins. Currently the integration tests with these plugins
pull in all of the plugins into the same test classpath. That can cause
problems when the dependencies of those repositories need to differ.
This commit splits the tests into separate qa subprojects.

### CODE REVIEW

File: x-pack/plugin/repository-encrypted/qa/azure/build.gradle [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+apply plugin: 'elasticsearch.internal-cluster-test'
+apply plugin: 'elasticsearch.java'
+
+dependencies {
+    internalClusterTestImplementation testArtifact(project(':plugins:repository-azure'), 'internalClusterTest')
+    internalClusterTestImplementation testArtifact(project(':x-pack:plugin:repository-encrypted'), 'test')
```
Comment (mark-vieira) [inline x-pack/plugin/repository-encrypted/qa/azure/build.gradle:7]: FWIW, in this scenario you can omit the `test` parameter as this is the default.

**Review by mark-vieira — APPROVED** [review]
LGTM. Is the parent `:qa` project build file strictly required? I don't _think_ it is. If not let's delete it. Even though it's empty, we still have to compile it.

File: x-pack/plugin/repository-encrypted/qa/azure/build.gradle (hunk 1 above)
Comment (rjernst) [reply x-pack/plugin/repository-encrypted/qa/azure/build.gradle:7]: I'll leave it for clarity, but feel free to remove it when I'm not looking

**Comment by mark-vieira** [comment]
> I believe it is necessary. The way we build out all of x-pack in settings.gradle is through recursively finding build.gradle files. Otherwise we wouldn't know qa was supposed to be a subproject.

Roger that.

**Pull request merged**